src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskAssignByIdControllerTest.java

```diff
@@ -201,323 +147,8 @@ public void assigner_has_grant_type_specific_and_permission_manage_own_can_assig
-
-        initiateTask(taskVariables, Jurisdiction.WA);
-
-        //first assign
-        common.setupLeadJudgeForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-        //second assign
-        common.setupLeadJudgeForSpecificAccess(secondAssigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void assigner_has_grant_type_specific_and_permission_read_manage_execute_cancel_can_assign_task() {
-        //assigner role : read, manage, execute, cancel
-        //assignee role : read, manage, execute, cancel
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "reviewSpecificAccessRequestLegalOps");
-        taskId = taskVariables.getTaskId();
-
-        common.setupCaseManagerForSpecificAccess(assignerCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-
-        initiateTask(taskVariables, Jurisdiction.WA);
-
-        common.setupCaseManagerForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void assigner_has_grant_type_specific_and_permission_read_manage_execute_cancel_can_assign_other_already_assigned_task() {
-        //assigner role : read, manage, execute, cancel
-        //assignee role : read, manage, execute, cancel
-        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "reviewSpecificAccessRequestLegalOps");
-        taskId = taskVariables.getTaskId();
-
-        common.setupCaseManagerForSpecificAccess(assignerCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-
-        initiateTask(taskVariables, Jurisdiction.WA);
-
-        //first assign
-        common.setupCaseManagerForSpecificAccess(assigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-
-        //second assign
-        common.setupCaseManagerForSpecificAccess(secondAssigneeCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
-        assignTaskAndValidate(taskVariables, getAssigneeId(assigneeCredentials.getHeaders()));
-        common.cleanUpTask(taskId);
-    }
-
-    @Test
-    public void assigner_has_grant_type_challenged_and_permission_manage_should_not_assign_a_task_to_assignee_with_read_permission() {
```

Comment:
Is there an IT for this?

Reply:
This is negative scenario for assignee permission. Covered in PostTaskAssignByIdControllerTest::should_return_a_403_when_the_assignee_does_not_have_execute_or_own_permissions. Also in CftQueryServiceAssignTaskTest::should_not_retrieve_a_task_when_grant_type_challenged_without_assigner_assignee_having_required_permission. In CftQueryServiceAssignTaskTest there are some negative permission tests for other grant types as well.

Reply:
make sure CftQueryServiceAssignTaskTest has got a test to cover assignee doesn't have right permission or add one